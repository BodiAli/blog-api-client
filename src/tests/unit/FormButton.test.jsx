import { render, screen } from "@testing-library/react";
import FormButton from "../../components/FormButton/FormButton";
import { expect } from "vitest";

describe("FormButton component", () => {
  test("Should render a button with classname and children prop", () => {
    const { container } = render(<FormButton>Form button</FormButton>);

    expect(container).toMatchSnapshot();
  });

  test("Should disable button when disabled is true", () => {
    render(<FormButton disabled={true}>Form button</FormButton>);

    expect(screen.getByRole("button")).toBeDisabled();
  });
});
