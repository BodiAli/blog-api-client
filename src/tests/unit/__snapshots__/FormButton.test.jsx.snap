// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormButton component > Should render a button with classname and children prop 1`] = `
<div>
  <button
    class="_button_64628a "
    type="submit"
  >
    Form button
  </button>
</div>
`;
